shell/common/shell.cc

```diff
@@ -1511,4 +1551,26 @@ std::shared_ptr<fml::SyncSwitch> Shell::GetIsGpuDisabledSyncSwitch() const {
   return is_gpu_disabled_sync_switch_;
 }
 
+std::unique_ptr<Shell> Shell::ShellHolder::makeShell() {
+  if (is_consumed_) {
```

Comment:
I'd prefer this was a function like this:
```c++
std::unique_ptr<Shell> makeShell(std::unique_ptr<ShellHolder> holder);
```

That way we can eliminate the `is_consumed_` property and you won't be left with the pointer to an object that has been "spent".